tests/unit/test_charm.py

```diff
@@ -116,54 +102,48 @@ def test__alerting_config(self):
             self.harness.charm._alerting_config()
             self.assertEqual(sorted(logger.output), ["DEBUG:charm:No alertmanagers available"])
 
+    @patch("ops.model.Container.can_connect", MagicMock(return_value=False))
     @patch("charm.LokiOperatorCharm._loki_config")
-    def test__on_config_cannot_connect(self, mock_loki_config, *unused):
+    def test__on_config_cannot_connect(self, mock_loki_config):
         self.harness.set_leader(True)
         mock_loki_config.return_value = yaml.safe_load(LOKI_CONFIG)
-        self.harness.charm._container.can_connect = Mock()
-        self.harness.charm._container.can_connect.return_value = False
-        self.harness.update_config(yaml.safe_load(LOKI_CONFIG))
+
+        # Since harness was not started with begin_with_initial_hooks(), this must
+        # be emitted by hand to actually trigger _configure()
+        self.harness.charm.on.config_changed.emit()
         self.assertIsInstance(self.harness.charm.unit.status, WaitingStatus)
 
-    @patch("ops.testing._TestingPebbleClient.push")
+    @patch("ops.testing._TestingPebbleClient.push", Mock())
+    @patch("ops.model.Container.can_connect", MagicMock(return_value=True))
```

Comment:
Haven't tried myself but I expected that these mocks should not be needed.

Reply:
`.push` mocks were already removed in a previous patch. I'll also remove `.can_connect`